store/store.go

```diff
@@ -499,6 +505,32 @@ func (s *Store[H]) get(ctx context.Context, hash header.Hash) ([]byte, error) {
 	return data, nil
 }
 
+// advanceContiguousHead return a new highest contiguous height
+// or returns the given height if not found.
+func (s *Store[H]) advanceContiguousHead(ctx context.Context, height uint64) {
+	newHead, ok := s.nextContiguous(ctx, height)
+	if ok && newHead.Height() > height {
+		s.contiguousHead.Store(&newHead)
+		s.heightSub.SetHeight(newHead.Height())
+		log.Infow("new head", "height", newHead.Height(), "hash", newHead.Hash())
+		s.metrics.newHead(newHead.Height())
+	}
+}
+
+func (s *Store[H]) nextContiguous(ctx context.Context, height uint64) (H, bool) {
+	var newHead H
+	newHeight := height
```

Comment:
readability optional uber nit: 

this var here is only to check whether the head advanced or not
another way of expressing this is to make `advanced` or `changed` var updated if change actually happens
this is arguably clearer

Reply:
Ah wait, do we actually need to return the bool here? It seems like no, as in advance method we do the same one: newHead > height

So i think we can simplify this part as well